tests/api_connexion/endpoints/test_dataset_endpoint.py

```diff
@@ -81,6 +84,95 @@ def _create_dataset(self, session):
         return dataset_model
 
 
+class TestDeleteDatasetEndpoint(TestDatasetEndpoint):
+    def test_delete_should_respond_204(self, session):
+        dataset_model = DatasetModel(
+            id=1,
+            uri="s3://bucket/key",
+            extra={"foo": "bar"},
+            created_at=timezone.parse(self.default_time),
+            updated_at=timezone.parse(self.default_time),
+        )
+
+        session.add(dataset_model)
+        session.commit()
+        datasets = session.query(DatasetModel).all()
+        assert len(datasets) == 1
+        response = self.client.delete(
+            f"/api/v1/datasets/{urllib.parse.quote('s3://bucket/key', safe='')}",
```

Comment:
Ew, escaping URI as one single path component is so ugly. But I guess we don’t have many choices here. We should probably document this in the Open API spec (or somewhere else?) how to put the URI in the API URL.

Reply:
Actually, I'm not sure why the datasets endpoint choose to use URI as it's parameter instead of `id` like for DAG endpoint so I'm just following the existing pattern. I agree that it may not be the best choice.

Reply:
I just found that previously we already use `id` in the endpoint but changed by https://github.com/apache/airflow/pull/25770. Added some relevant docs in the spec on how to put the URI in API URL. Re-requesting your review @uranusjr